src/components/TodoList.tsx

```diff
@@ -0,0 +1,22 @@
+import { Todo } from '../types/Todo';
+import React, { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TempTodoItem } from './TempTodoItem';
+import { TodoListContext } from '../variables/LangContext';
+
+type Props = {
+  visibleTodos: Todo[];
+};
+
+export const TodoList: React.FC<Props> = ({ visibleTodos }) => {
+  const { tempTodo } = useContext(TodoListContext);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {visibleTodos.map(todo => (
+        <TodoItem key={todo.id} todo={todo} />
+      ))}
+      {tempTodo && <TempTodoItem tempTodo={tempTodo} />}
+    </section>
```

Comment:
```suggestion
    <section className="todoapp__main" data-cy="TodoList">
      {visibleTodos.map(todo => (
        <TodoItem key={todo.id} todo={todo} />
      ))}
      {tempTodo && <TodoItem todo={tempTodo} />}
    </section>
```
it can be the same todo as others, but just with loader(you can add the prop isLoading)